### PR TITLE
Ignore wstring_convert and codecvt_utf8 deprecation error

### PR DESCRIPTION
**Pull request by jianlingzhong**

### CODE REVIEW

File: compiler+runtime/src/cpp/jank/read/parse.cpp [hunk 1]
```diff
@@ -2,7 +2,10 @@
 
 #include <fmt/format.h>
 
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
 #include <jank/native_persistent_string/fmt.hpp>
+#pragma clang diagnostic pop
```
Comment (jeaye) [inline compiler+runtime/src/cpp/jank/read/parse.cpp:8]: Why is this one needed? If there's a deprecated usage in that header, we should ignore it in that header, not at once inclusion point. This header is included in dozens of places.

File: compiler+runtime/src/cpp/jank/util/string_builder.cpp [hunk 2]
```diff
@@ -1,6 +1,9 @@
 #include <bit>
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
 #include <codecvt>
 #include <locale>
+#pragma clang diagnostic pop
```
Comment (jeaye) [inline compiler+runtime/src/cpp/jank/util/string_builder.cpp:6]: Is the entire header deprecated? I thought it was only `wstring_convert`.

File: compiler+runtime/src/cpp/jank/read/parse.cpp (hunk 1 above)
Comment (jianlingzhong) [reply compiler+runtime/src/cpp/jank/read/parse.cpp:8]: adding this in this single place fixed it. without it i still got complains on wstring_convert:

```
❯ bin/compile
[8/32] Building CXX object CMakeFiles/jank_lib.dir/src/cpp/jank/read/parse.cpp.o
FAILED: CMakeFiles/jank_lib.dir/src/cpp/jank/read/parse.cpp.o
/Users/jank/compiler+runtime/build/llvm-install/usr/local/bin/clang++ -DBOOST_ATOMIC_DYN_LINK -DBOOST_ATOMIC_NO_LIB -DBOOST_FILESYSTEM_DYN_LINK -DBOOST_FILESYSTEM_NO_LIB -DBOOST_SYSTEM_DYN_LINK -DBOOST_SYSTEM_NO_LIB -DFMT_SHARED -D__STDC_CONSTANT_MACROS -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/Users/jank/compiler+runtime/build -I/Users/jank/compiler+runtime -I/Users/jank/compiler+runtime/include/cpp -isystem /Users/jank/compiler+runtime/build/llvm-install/usr/local/include -isystem /Users/jank/compiler+runtime/third-party/nanobench/include -isystem /Users/jank/compiler+runtime/third-party/folly -isystem /Users/jank/compiler+runtime/third-party/bpptree/include -isystem /Users/jank/compiler+runtime/third-party/bdwgc/include -isystem /Users/jank/compiler+runtime/third-party/immer -isystem /Users/jank/compiler+runtime/third-party/magic_enum/include/magic_enum -isystem /Users/jank/compiler+runtime/third-party/cli11/include -isystem /Users/jank/compiler+runtime/third-party/fmt/include -isystem /Users/jank/compiler+runtime/third-party/libzippp/src -isystem /opt/homebrew/include -isystem /opt/homebrew/Cellar/openssl@3/3.4.0/include -fPIC -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -Wimplicit-fallthrough -Wcovered-switch-default -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wstring-conversion -Wmisleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -g -std=gnu++20 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.2.sdk -fcolor-diagnostics -std=gnu++20 -DIMMER_HAS_LIBGC=1 -DIMMER_TAGGED_NODE=0 -DHAVE_CXX14=1 -I/opt/homebrew/include -Werror -DJANK_DEBUG -Wall -Wextra -Wpedantic -Wfloat-equal -Wuninitialized -Wswitch-enum -Wnon-virtual-dtor -Wold-style-cast -Wno-gnu-case-range -Wno-gnu-conditional-omitted-operand -Wno-implicit-fallthrough -Wno-covered-switch-default -Wno-invalid-offsetof -fno-common -fno-rtti -fexceptions -O0 -DJANK_VERSION=\"0.1-4c7a53cbf2f451a73eba7e16d6372673828ddd38\" "-DJANK_JIT_FLAGS=\"-std=gnu++20 -DIMMER_HAS_LIBGC=1 -DIMMER_TAGGED_NODE=0 -DHAVE_CXX14=1 -I/opt/homebrew/include -Werror -DJANK_DEBUG -w\"" -DJANK_CLANG_PREFIX=\"/Users/jank/compiler+runtime/build/llvm-install/usr/local\" -MD -MT CMakeFiles/jank_lib.dir/src/cpp/jank/read/parse.cpp.o -MF CMakeFiles/jank_lib.dir/src/cpp/jank/read/parse.cpp.o.d -o CMakeFiles/jank_lib.dir/src/cpp/jank/read/parse.cpp.o -c /Users/jank/compiler+runtime/src/cpp/jank/read/parse.cpp
In file included from /Users/jank/compiler+runtime/src/cpp/jank/read/parse.cpp:5:
In file included from /Users/jank/compiler+runtime/include/cpp/jank/native_persistent_string/fmt.hpp:5:
In file included from /Users/jank/compiler+runtime/include/cpp/jank/native_persistent_string.hpp:5:
In file included from /Users/jank/compiler+runtime/include/cpp/jank/type.hpp:3:
In file included from /Users/jank/compiler+runtime/build/llvm-install/usr/local/bin/../include/c++/v1/map:2175:
In file included from /Users/jank/compiler+runtime/build/llvm-install/usr/local/bin/../include/c++/v1/functional:540:
In file included from /Users/jank/compiler+runtime/build/llvm-install/usr/local/bin/../include/c++/v1/__functional/boyer_moore_searcher.h:27:
In file included from /Users/jank/compiler+runtime/build/llvm-install/usr/local/bin/../include/c++/v1/vector:2984:
/Users/jank/compiler+runtime/build/llvm-install/usr/local/bin/../include/c++/v1/locale:3280:1: error: 'wstring_convert<std::codecvt_utf8<char32_t>, char32_t>' is deprecated [-Werror,-Wdeprecated-declarations]
 3280 | wstring_convert<_Codecvt, _Elem, _WideAlloc, _ByteAlloc>::to_bytes(const _Elem* __frm, const _Elem* __frm_end) {
      | ^
/Users/jank/compiler+runtime/build/llvm-install/usr/local/bin/../include/c++/v1/locale:3179:70: note: in instantiation of member function 'std::wstring_convert<std::codecvt_utf8<char32_t>, char32_t>::to_bytes' requested here
 3179 |   _LIBCPP_HIDE_FROM_ABI byte_string to_bytes(_Elem __wchar) { return to_bytes(&__wchar, &__wchar + 1); }
      |                                                                      ^
/Users/jank/compiler+runtime/src/cpp/jank/read/parse.cpp:50:58: note: in instantiation of member function 'std::wstring_convert<std::codecvt_utf8<char32_t>, char32_t>::to_bytes' requested here
   50 |       native_persistent_string const converted(converter.to_bytes(codepoint));
      |                                                          ^
/Users/jank/compiler+runtime/build/llvm-install/usr/local/bin/../include/c++/v1/locale:3137:28: note: 'wstring_convert<std::codecvt_utf8<char32_t>, char32_t>' has been explicitly marked deprecated here
 3137 | class _LIBCPP_TEMPLATE_VIS _LIBCPP_DEPRECATED_IN_CXX17 wstring_convert {
      |                            ^
/Users/jank/compiler+runtime/build/llvm-install/usr/local/bin/../include/c++/v1/__config:747:41: note: expanded from macro '_LIBCPP_DEPRECATED_IN_CXX17'
  747 | #    define _LIBCPP_DEPRECATED_IN_CXX17 _LIBCPP_DEPRECATED
      |                                         ^
/Users/jank/compiler+runtime/build/llvm-install/usr/local/bin/../include/c++/v1/__config:712:49: note: expanded from macro '_LIBCPP_DEPRECATED'
  712 | #      define _LIBCPP_DEPRECATED __attribute__((__deprecated__))
      |                                                 ^
1 error generated.
[19/32] Building CXX object CMakeFiles/jank_lib.dir/src/cpp/jank/codegen/llvm_processor.cpp.o
ninja: build stopped: subcommand failed.
```

File: compiler+runtime/src/cpp/jank/util/string_builder.cpp (hunk 2 above)
Comment (jianlingzhong) [reply compiler+runtime/src/cpp/jank/util/string_builder.cpp:6]: had to add those around the header as well. otherwise still got the complains.

**Comment by jeaye** [comment]
Thanks for fixing this!

**Pull request merged**